pkg/collector/corechecks/snmp/internal/checkconfig/profile_json_bundle.go

```diff
@@ -0,0 +1,89 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package checkconfig
+
+import (
+	"compress/gzip"
+	"encoding/json"
+	"errors"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-agent/pkg/networkdevice/profile/profiledefinition"
+)
+
+func loadBundleJSONProfiles() (profileConfigMap, error) {
+	jsonStr, err := getProfilesBundleJSON()
+	if err != nil {
+		return nil, err
+	}
+
+	profiles, err := unmarshallProfilesBundleJSON(jsonStr)
+	if err != nil {
+		return nil, err
+	}
+	return profiles, nil
+}
+
+func unmarshallProfilesBundleJSON(jsonStr []byte) (profileConfigMap, error) {
+	bundle := profiledefinition.ProfileBundle{}
+	err := json.Unmarshal(jsonStr, &bundle)
+	if err != nil {
+		return nil, err
+	}
+
+	profiles := make(profileConfigMap)
+	for _, profile := range bundle.Profiles {
+		if profile.Profile.Name == "" {
+			log.Warnf("Profile with missing name: %s", profile.Profile.Name)
+			continue
+		}
+
+		if _, exist := profiles[profile.Profile.Name]; exist {
+			log.Warnf("duplicate profile found: %s", profile.Profile.Name)
+			continue
```

Comment:
@NouhaManai96

It's not definitive yet, but I think we can likely resolve that at backend level when the downloaded `profiles.json.gz` is created.

I mean, if in the future the profiles.json.gz will contain both default and customer profiles, for a specific profile, the backend will only send the one that has priority (customer profile has priority over default profile).

So, in theory, the two verifications above (empty name & duplicate names) shouldn't happen.

That said, we can possibly add a `isUserProfile` check as protective measure. I'm having a look if there is a better way to implement this.

Reply:
> what about the case where customers choose the profile they want to monitor (I mean by specifying the profile in the instance config) ?
not sure how often we'd expect this to happen (or if ever) but if a customer creates a custom profile but for some reason decide they want to use the default profile instead, this will discard the default in favour of the custom

@NouhaManai96 I think this is not a expected use case, even today with yaml profiles in `profiles` and `default_profiles` folder we don't support this use case.

If a user create a custom/user profile with the same name as default OOTB profiles it will take precedence whether the profile is 1/ autodetected/matched via sysObjectID or 2/ specified the profile explicitely in the instance config.

Reply:
makes sense if this is not supported in general 👍